### ISSUE TITLE
For first timers there is a missing step

### ISSUE THREAD

**Issue by Chris-Marassovich** [MicrosoftDocs/xamarin-docs#83]
If you are a first timer like myself and thing you can do "file->new" and follow this doco it will not.  
You get a nice null reference exception.

There is a missing step around the use of the DependencyService.  Simple to fix just visit the https://docs.microsoft.com/en-au/xamarin/xamarin-forms/app-fundamentals/dependency-service/introduction. Took me a little while to chase this one down and I hope this saves others some time.

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3ab6098a-39c7-151f-124b-f88aa62c4659
* Version Independent ID: dc27c1e4-ff12-c1af-646a-c2617def0bcd
* Content: [Local Databases - Xamarin](https://docs.microsoft.com/en-au/xamarin/xamarin-forms/app-fundamentals/databases#using-sqlite)
* Content Source: [docs/xamarin-forms/app-fundamentals/databases.md](https://github.com/MicrosoftDocs/xamarin-docs/blob/live/docs/xamarin-forms/app-fundamentals/databases.md)
* Service: **unspecified**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

**Comment by conceptdev** [comment]
@Chris-Marassovich which part of the Dependency Injection doc did you find most helpful, and believe should be covered in the Xamarin.Forms Database doc?

**Comment by Chris-Marassovich** [comment]
@conceptdev I can see why you need to confirm with me ... I made quite a few typo's in that first comment.
My point I was trying to get across was this:  While the Database doc appeared to be straight forward that you could just do a file->new and take the code at it works! there is a sneaky step that is hard to spot.  It is the use of dependency service.  At compile you get a null ref error and it is not obvious that you have missed setting up the dependency service.
I am not sure copying the doco is the right solution but at least make a mention of it along with a link to it so the requirement is obvious.

It is frustrating when learning something and you have taken what appears to be basic code examples and it does not work.  If I am learning Xamarin forms and local database for the first time I will also not be aware of the dependency service.  After struggling with this the dependency service is obvious but now I know after a lot of effort.

I am trying to save others the time I lost.  Maybe this is not the right place in the doco for that??

**Comment by swinterrowd** [comment]
I have been looking at this ToDo code for a day or so now and am not impressed with it as an introduction to using a local database with Xamarin Forms. I am attempting to learn the basics of Xamarin Forms with a sqlite database. I am considering changing a WPF project I wrote to run on an Android. I am also trying to determine if Xamarin Forms or Xamarin is the platform I should use. I was under the impression that this sample was supposed to be a simple introduction about how to do this. There does not appear to be any other examples available. As an introduction, this code is to complex. The sample program has way more code than what is needed to grasp the required concepts. There are many modifications and commented out sections in the sample code that make it more difficult to follow than it has to be. For example, what's with the whole text to speech thing and how does that fit into the basic concepts of trying to use a local database? If this code is is supposed to be for learning how to use a local database with Xamarin Forms, dumb it down. Don't make it more complex than it has to be. There is plenty to try and figure out without additional and seemingly needless complexity.

**Comment by conceptdev** [comment]
Thank you both for your feedback - it's very helpful to hear different points of view. If there was a quick/sample change, I'd make it now (still considering a quick update to link to dependency service, to at least address Chris' feedback), but I've added a task to our backlog to look at the entire doc & sample. I trust we come up with something better for everyone. Thanks again!

**Comment by githubabu** [comment]
I'm new to Xamarin and I am struggling with this. I've tried creating a new app and replicating the code but have reached a blocking point and I can't see on your code what it is that I'm missing. I have  a database with an ID and name, I removed the text to speech code. Whenever my code gets to the createTableAsync line to create the table it crashes and doesn't create the table. I just can't see what I'm missing, I've copied everything almost identically, do I need to add anything to the iOS project?
